### PR TITLE
NuttX upgrade for V1.7 Release 

### PR DESCRIPTION
**Pull request by davids5**
This PR will provide the NuttX upgrade for V1.7 Release See https://github.com/PX4/Firmware/issues/7852 

This PR supersede https://github.com/PX4/Firmware/pull/7428 and is the continuation of  https://github.com/PX4/Firmware/pull/6438 - https://github.com/PX4/Firmware/pull/6438 will remain open for progress tracking of this branch with respect to NXP flight V3 progress that will continue here in.

### CODE REVIEW

**Comment by davids5** [comment]
Jenkins test this please.

**Comment by davids5** [comment]
Jenkins test this please.

**Comment by davids5** [comment]
rebased on master @ 38a4c1e90b8c77f2d10f9ae559c28b6fd4578f98

**Comment by davids5** [comment]
@PX4TestFlights - I need get a baseline pulse on this. 

Please build px4fmu-v3_default. Then safe bench test and if OK then fly this on:
PX4FMUv2 hw no errata 
PX4FMUv2 mini.

If you have a serial based companion computer on a nuttx based build that would also be good test vehicle.

**Comment by santiago3dr** [comment]
couple flights:
pixhawk 1 no errata:
https://logs.px4.io/plot_app?log=51b90ce3-a44d-411a-a10c-6f6af0ac3303

pixhawk mini:
https://logs.px4.io/plot_app?log=badcea93-df76-4f53-9159-86c1d09cc2bb

flights were good; ran through all usual flight modes; didn't encounter any issues

**Comment by r0gelion** [comment]
@PX4TestFlights unassigned, reassign when needed.

**Comment by davids5** [comment]
@r0gelion - 

1. does test have a setup with a companion computer (serial connected), that can be tested with one of the nuttx based board builds? 
2. does test have a setup any serial connected device with hardware handshaking?

**Comment by r0gelion** [comment]
Hi @davids5 ,

> does test have a setup with a companion computer (serial connected), that can be tested with one of the nuttx based board builds?

No, we don't have a setup with a companion computer, what's your recommendation for this setup?

> does test have a setup any serial connected device with hardware handshaking?

No, we don't have a setup with a serial device with handshaking connected, what's your recommendation for this setup?

**Comment by davids5** [comment]
@LorenzMeier, @dagar Would have some suggestions for @r0gelion on how we can get good test coverage on high-speed Serial with handshaking?

**Comment by dagar** [comment]
@r0gelion which telemetry radios do you use for pixhawk 3 pro testing?

**Comment by r0gelion** [comment]
> which telemetry radios do you use for pixhawk 3 pro testing?

3DR radios

**Comment by davids5** [comment]
@r0gelion Please test this on aerofc - looking for any mavlink related issue.

**Comment by davids5** [comment]
@r0gelion - @dagar will provide some 3dr radio set up test cases as well.

**Comment by santiago3dr** [comment]
couple flights with the intel aero;
indoor test:
https://logs.px4.io/plot_app?log=59d4bcdb-ec4b-4936-a5ff-93016908405e

full flight:
https://logs.px4.io/plot_app?log=604b5d72-e41a-43f1-8ce8-97334d57cbe3

a lot of minor issues that are also present with master

**Comment by davids5** [comment]
@dagar rebased on master @ 0780e430fda5396a6ab64380380a993afabb2015

This has the broken PX4IO on V3, that came in from master

**Comment by dagar** [comment]
@santiago3dr are you still doing logging to QGC over the telem radio or do you download the log from the sd card?

**Comment by r0gelion** [comment]
@dagar we download the logs through scp (WinSCP)

**Comment by davids5** [comment]
rebased on Master at 3c18be387cc47c9875137745b171e7ccec901f2e

**Comment by davids5** [comment]
Tied to nuttx, apps  px4_firmware_nuttx-master 

rebased on upstream nuttx master @ 3647ace55958c094cfe93f07092d308a768f5ca3
rebased on upstream apps master @ 21c9b793e2ca6f4e86388c80d7af5c466de5e71c

File: src/drivers/boards/px4fmu-v3/px4fmu_init.c [hunk 1]
```diff
@@ -119,17 +119,10 @@ __END_DECLS
 /****************************************************************************
  * Private Data
  ****************************************************************************/
-#if defined(BOARD_HAS_SIMPLE_HW_VERSIONING)
-static int hw_version = 0;
-static int hw_revision = 0;
+static int hw_version = -1;
```
Comment (davids5) [inline src/drivers/boards/px4fmu-v3/px4fmu_init.c]: @dagar - there was an API change.
```
* Returned Value:
 *   An integer value of this boards hardware version.
 *   A value of -1 is the default for boards not supporting the BOARD_HAS_VERSIONING API.
 *   A value of 0 is the default for boards supporting the API but not having version.
```

Comment (dagar) [reply src/drivers/boards/px4fmu-v3/px4fmu_init.c]: Ah good catch. I went back to master and redid these with unifdef.

Comment (dagar) [reply src/drivers/boards/px4fmu-v3/px4fmu_init.c]: Fixed

**Pull request merged**